schemas/dart_model.schema.json

```diff
@@ -0,0 +1,76 @@
+{
+  "$id": "https://github.com/dart-lang/macros/blob/main/schemas/dart_model.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "oneOf": [
+    {"$ref": "#/$defs/Model"}
+  ],
+  "$defs": {
+    "Annotation": {
+      "type": "object",
+      "description": "An annotation.",
+      "properties": {
+        "type": {"$ref": "#/$defs/QualifiedName"}
+      }
+    },
+    "Interface": {
+      "type": "object",
+      "description": "An interface.",
+      "properties": {
+        "annotations": {
+          "type": "array",
+          "items": {"$ref": "#/$defs/Annotation"}
+        },
+        "members": {
+          "type": "object",
+          "description": "Map of members by name.",
+          "additionalProperties": {"$ref": "#/$defs/Member"}
+        },
+        "properties": {"$ref": "#/$defs/Properties"}
+      }
+    },
+    "Member": {
+      "type": "object",
+      "description": "Member of a scope.",
+      "properties": {
+        "properties": {"$ref": "#/$defs/Properties"}
+      }
+    },
+    "Model": {
+      "type": "object",
+      "properties": {
+        "uris": {
+          "type": "object",
+          "description": "Libraries by URI.",
+          "additionalProperties": {"$ref": "#/$defs/Library"}
+        }
+      }
+    },
+    "Properties": {
+      "type": "object",
+      "description": "Set of boolean properties.",
+      "properties": {
+        "isAbstract": {"type": "boolean"},
+        "isClass": {"type": "boolean"},
+        "isGetter": {"type": "boolean"},
+        "isField": {"type": "boolean"},
+        "isMethod": {"type": "boolean"},
+        "isStatic": {"type": "boolean"},
+        "isSynthetic": {"type": "boolean"}
+      }
+    },
+    "Library": {
+      "description": "Library.",
+      "type": "object",
+      "properties": {
+        "scopes": {
```

Comment:
It seems weird to call this scopes and not just declarations, which could have the regular top level declarations, which may or may not introduce a "scope". Or do we intend to model top level declarations like fields/functions as "scopes" also? It might make sense, but then you also end up wanting to probably be able to model lambda scopes, block scopes, etc....

Reply:
Functions definitely introduce a scope - their parameters should be resolved in the function scope (so that they can resolve to the parameters and not top level members, as an example). So, definitely more than just interfaces are at play here.

Reply:
Yes, good point. I was excluding those from "Scope", so it really isn't a good name.

I expect "Interface" to be followed by more types.

Let's pull the discussion on this and "uris" to #10